timesketch/lib/utils.py

```diff
@@ -101,79 +117,59 @@ def read_and_validate_csv(file_handle, delimiter=','):
     if not isinstance(delimiter, six.text_type):
         delimiter = codecs.decode(delimiter, 'utf8')
 
-    # Due to issues with python2.
-    if six.PY2:
-        delimiter = str(delimiter)
-
-    reader = csv.DictReader(file_handle, delimiter=delimiter)
-    csv_header = reader.fieldnames
-    missing_fields = []
-    # Validate the CSV header
-    for field in mandatory_fields:
-        if field not in csv_header:
-            missing_fields.append(field)
-    if missing_fields:
-        raise RuntimeError(
-            'Missing fields in CSV header: {0:s}'.format(
-                ','.join(missing_fields)))
     try:
-        for row in reader:
-            # There is a condition in which the CSV reader can read a
-            # single lines as multiple, causing issues with importing.
-            # TODO: Swap the CSV library for the user of pandas.
-            if not row:
-                continue
-            if not row['datetime']:
-                logger.warning(
-                    'Row missing a datetime object, skipping [{0:s}]'.format(
-                        ','.join([str(x).replace(
-                            '\n', '').strip() for x in row.values()])))
-                continue
-            try:
-                # normalize datetime to ISO 8601 format if it's not the case.
-                parsed_datetime = parser.parse(row['datetime'])
-                row['datetime'] = parsed_datetime.isoformat()
-
-                normalized_timestamp = int(
-                    time.mktime(parsed_datetime.utctimetuple()) * 1000000)
-                normalized_timestamp += parsed_datetime.microsecond
-                row['timestamp'] = str(normalized_timestamp)
-                if 'tag' in row:
-                    row['tag'] = [x for x in _parse_tag_field(row['tag']) if x]
-
-                _scrub_special_tags(row)
-            except ValueError:
-                continue
-
-            yield row
-    except csv.Error as e:
+        reader = pandas.read_csv(file_handle, sep=delimiter)
```

Comment:
I would opt not do it this way..I would rather use

```
try:
   for df_chunk in pd.read_csv(file_handle, sep=delimiter, chunksize=SOME_DEFAULT_VALUE):
      # Do all the validation, etc....
```

That is to read in the file in chunks, that way you are not reading in the entire file in memory at time, preventing us trying to read the entire content of a very large file, rather to split it up in chunks, we are doing a generator here, so it would be better